src/dbnode/client/session_test.go

```diff
@@ -341,6 +319,89 @@ func TestSeriesLimit(t *testing.T) {
 	require.NoError(t, sess.Close())
 }
 
+//nolint:dupl
+func TestSeriesLimit_FetchTaggedIDs(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// mock the host queue to return a result with a single series, this results in 3 series total, one per shard.
+	sess := setupMultipleInstanceCluster(t, ctrl, func(op op, host topology.Host) {
+		fOp := op.(*fetchTaggedOp)
+		assert.Equal(t, int64(2), *fOp.request.SeriesLimit)
+		shardID := strings.Split(host.ID(), "-")[2]
+		op.CompletionFn()(fetchTaggedResultAccumulatorOpts{
+			host: host,
+			response: &rpc.FetchTaggedResult_{
+				Exhaustive: true,
+				Elements: []*rpc.FetchTaggedIDResult_{
+					{
+						// use shard id for the metric id so it's stable across replicas.
+						ID: []byte(shardID),
+					},
+				},
+			},
+		}, nil)
+	})
+
+	iter, meta, err := sess.fetchTaggedIDsAttempt(context.TODO(), ident.StringID("ns"),
+		index.Query{Query: idx.NewAllQuery()},
+		index.QueryOptions{
+			// set to 6 so we can test the instance series limit is 2 (6 /3 instances per replica * InstanceMultiple)
+			SeriesLimit:      6,
+			InstanceMultiple: 1,
+		})
+	require.NoError(t, err)
+	require.NotNil(t, iter)
+	// expect a series per shard.
+	require.Equal(t, 3, iter.Remaining())
+	require.True(t, meta.Exhaustive)
+	require.NoError(t, sess.Close())
+}
+
+//nolint:dupl
+func TestSeriesLimit_Aggregate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// mock the host queue to return a result with a single series, this results in 3 series total, one per shard.
+	sess := setupMultipleInstanceCluster(t, ctrl, func(op op, host topology.Host) {
+		aOp := op.(*aggregateOp)
+		assert.Equal(t, int64(2), *aOp.request.SeriesLimit)
```

Comment:
really this is the assertion we care about right?

Reply:
This verifies the correct limit was sent. The assertion at the end verifies the results are properly accumulated across shards, with 1 result from each shard for a total of 3